app/javascript/src/ActiveDocs/ThreeScaleApiDocs.ts

```diff
@@ -17,6 +20,96 @@ const appendSwaggerDiv = (container: HTMLElement, id: string): void => {
+type BodyValue = boolean | number | string | { [key: string]: BodyValue }
+
+/**
+ * Transforms an object into form data representation, also URL-encoding the values,
+ * Example:
+ * {
+ *   a_string: 'hello',
+ *   an_array: [
+ *     { first: 1 },
+ *     { second: 1, extra_param: 'with whitespace'}
+ *   ]
+ * }
+ * =>
+ * {
+ *   a_string: 'hello',
+ *   'an_array[0][first]': '1',
+ *   'an_array[1][second]': '1',
+ *   'an_array[1][extra_param]': 'with%20whitespace'
+ * }
+ * @param object
+ */
+export const objectToFormData = (object: BodyValue): Record<string, boolean | number | string> => {
+  const buildFormData = (formData: Record<string, boolean | number | string>, data: BodyValue, parentKey?: string) => {
+    if (data && typeof data === 'object') {
+      Object.keys(data).forEach((key: string) => {
+        buildFormData(formData, data[key], parentKey ? `${parentKey}[${key}]` : key)
+      })
+    } else {
+      if (parentKey) {
+        formData[parentKey] = data ? data : ''
+      }
+    }
+  }
+  const formData = {}
+  buildFormData(formData, object)
+  return formData
+}
+
+/**
+ * Transforms the request body of the Service Management API Report, as
+ * SwaggerUI (or rather swagger-js) does not serialize arrays of objects properly
+ * The hack is to process the request body in two steps:
+ * 1. normalize the 'transactions' array, as its elements may be either an object (if the value is taken
+ *    from the example in the spec), or as a serialized JSON (if the field is changed manually)
+ * 2. "flatten" the objects by transforming them into form-data structure with the entries like
+ *    'transactions[0][app_id]': 'example'
+ *    'transactions[0][usage][hits]': 1
+ * @param body BackendApiReportBody
+ */
+export const transformReportRequestBody = (body: BackendApiReportBody): Record<string, boolean | number | string> => {
+  if (Array.isArray(body.transactions)) {
+    body.transactions = body.transactions.map(transaction => {
+      switch (typeof transaction) {
+        case 'object':
+          return transaction
+        case 'string':
+          try {
+            return JSON.parse(transaction) as BackendApiTransaction
+          } catch (error: unknown) {
+            return null
+          }
+        default:
+          return null
+      }
+    }).filter(element => element != null) as BackendApiTransaction[]
+  }
+  return objectToFormData(body as BodyValue)
+}
+
+const RequestBodyTransformerPlugin: SwaggerUIPlugin = () => {
+  return {
+    fn: {
+      execute: (req: ExecuteData): unknown => {
+        if (req.contextUrl.includes('api_docs/services/service_management_api.json')
+            && req.operationId === 'report'
+            && req.requestBody) {
+          req.requestBody = transformReportRequestBody(req.requestBody as BackendApiReportBody)
+        }
+        // eslint-disable-next-line @typescript-eslint/no-unsafe-call
+        return execute(req)
```

Comment:
If I understand this correctly, this replaces the original execute method to the one defined here, and this one at the end calls the original one, is that right?

Reply:
Yes, that's correct.